cyfra-rtrp/src/main/scala/io/computenode/cyfra/rtrp/surface/SurfaceManager.scala

```diff
@@ -0,0 +1,168 @@
+
+    surfaceFactory
+      .createSurface(window, config)
+      .map: surface =>
+        activeSurfaces(window.id) = surface
+        surfaceConfigs(window.id) = config
+
+        surface
+          .getCapabilities()
+          .foreach: capabilities =>
+            fireEvent(SurfaceEvent.SurfaceCreated(window.id, surface.id, capabilities))
+
+        surface
+
+  def createSurfaces(windows: List[Window], config: SurfaceConfig = SurfaceConfig.default): Try[List[RenderSurface]] =
+    val results = windows.map(createSurface(_, config))
+
+    val failures = results.collect { case Failure(ex) => ex }
+    if failures.nonEmpty then
+      results.collect { case Success(surface) => destroySurface(surface.windowId) }
+      Failure(new RuntimeException(s"Failed to create ${failures.size} surfaces"))
+    else Success(results.collect { case Success(surface) => surface })
+
+  def getSurface(windowId: WindowId): Option[RenderSurface] =
+    activeSurfaces.get(windowId)
+
+  def getActiveSurfaces(): Map[WindowId, RenderSurface] = activeSurfaces.toMap
+
+  def getSurfaceConfig(windowId: WindowId): Option[SurfaceConfig] =
+    surfaceConfigs.get(windowId)
+
+  def updateSurfaceConfig(windowId: WindowId, newConfig: SurfaceConfig): Try[Unit] =
+    getSurface(windowId) match
+      case Some(surface) =>
+        surfaceConfigs(windowId) = newConfig
+        // Note: Actual surface reconfiguration would happen in swapchain recreation
+        Success(())
+      case None =>
+        Failure(new IllegalArgumentException(s"No surface found for window $windowId"))
+
+  def destroySurface(windowId: WindowId): Try[Unit] =
+    activeSurfaces.remove(windowId) match
+      case Some(surface) =>
+        surfaceConfigs.remove(windowId)
+
+        val result = surface.destroy()
+
+        fireEvent(SurfaceEvent.SurfaceDestroyed(windowId, surface.id))
+
+        result
+      case None =>
+        Success(())
+
+  def handleWindowEvent(event: WindowEvent): Try[Unit] = Try:
+    event match
+      case WindowEvent.Resized(windowId, width, height) =>
+        activeSurfaces
+          .get(windowId)
+          .foreach: surface =>
+            surface.resize(width, height) match
+              case Success(_) =>
+                // Surface resized successfully, capabilities might have changed
+                surface
+                  .getCapabilities()
+                  .foreach: newCapabilities =>
+                    // Fire capabilities changed event (we'd need to compare with old capabilities)
+                    fireEvent(
+                      SurfaceEvent.SurfaceCapabilitiesChanged(
+                        windowId,
+                        surface.id,
+                        newCapabilities,
+                        newCapabilities, // TODO: track old capabilities
+                      ),
+                    )
+
+              case Failure(ex) =>
+                println(s"Warning: Failed to resize surface for window $windowId: ${ex.getMessage}")
```

Comment:
This could be rewritten as a for loop - There are two levels of nested foreach. You will need to use some methods of Try to handle the error.